### DISCUSSION TITLE
Could patched dependencies be more local-development friendly?

### DISCUSSION THREAD

**Discussion by TravisJRyan** [pnpm/pnpm#4946] [Ideas]
The new patched dependencies feature in 7.4.0 is great. To my understanding, it's the only way of directly editing node_modules for local development debugging.

As a developer, I find it somewhat cumbersome to make patches that I want to undo for local testing. Would there be a way to make this easier for strictly local development if you're sure you don't want to commit patches to package.json, lockfile and source control? Maybe some `pnpm clear patches` that resets everything?

**Comment by zkochan** [comment]
Why do you need patches for that at all? Just edit the files inside node_modules.

**Reply by TravisJRyan** [reply]
@zkochan Is this encouraged for local development? From my testing, it seems that this edits the files in the store and can cause unintuitive issues in other projects if not cleaned up. Or even if you always remember to undo your edits, there could be scenarios where you want to patch a dependency to do some testing with one project while running other projects that want the unaltered dependency.

**Reply by zkochan** [reply]
if you're on linux, the store is not corrupted. 

But even if it is, on next install it is fixed.

**Reply by zkochan** [reply]
also, you can use copying instead of hard linking
https://pnpm.io/npmrc#package-import-method

**Reply by TravisJRyan** [reply]
> if you're on linux, the store is not corrupted.
> 
> But even if it is, on next install it is fixed.

I tested with a simple example. It seems editing a module will corrupt the dependency for other projects (same hard link), but moving forward new installs will be correct. So if a user has a number of projects and edits node_modules in one for some local testing, they will need to either remember to undo their changes or delete node_modules and re-install in all these other projects, since these other projects retain the old hard link.


> also, you can use copying instead of hard linking https://pnpm.io/npmrc#package-import-method

This makes sense - I was hoping to preserve the performance benefits of hard linking. My understanding is there's no way around the above issue with hard-linking aside from patching during local development.
